db/migrate/20250130191533_add_email_confirmation_to_users.rb

```diff
@@ -0,0 +1,9 @@
+class AddEmailConfirmationToUsers < ActiveRecord::Migration[7.2]
+  def change
+    add_column :users, :unconfirmed_email, :string
+    add_column :users, :email_confirmation_token, :string
+    add_column :users, :email_confirmation_sent_at, :datetime
```

Comment:
We should be able to delete `:email_confirmation_sent_at` and just let the `generates_token_for` expiry handle the expiration validation (see my comment above about removing the `confirm_email_change` method)